metadata-ingestion/src/datahub/ingestion/api/source_helpers.py

```diff
@@ -59,6 +62,39 @@ def auto_workunit(
             yield item.as_workunit()
 
 
+def create_dataset_props_patch_builder(
+    datahub_dataset_name: str,
+    dataset_urn: str,
+    table: Any,  # TODO: refactor to avoid circular import while type-hinting with BaseTable
```

Comment:
you can use `if TYPE_CHECKING` - we do that in a couple other places to avoid this issue

Also, I had been thinking that this would take in the DatasetPropertiesClass aspect and convert that to patch. That way, it wouldn't need to take in the dataset name and custom properties as separate args